doc/RFCs/APInotification.md

```diff
@@ -0,0 +1,245 @@
+
+le `body` comporte le message/payload de la notification
+
+les headers `from` et `to` servent à identifier les composants du backend (from) et du frontend (to) concernés par la notification.
+
+le header `type` permet d'identifier immédiatement le type de notification : event, info, feedback, warning, teaser, error, alert, etc.
+
+le header `reference` pourrait servir à lier une notification à un évènement ou une action initiée par le front afin que celui identifie le feedback du backend sur le déroulement de cette action.
+
+le header `timestamp` est le timestamp unix du moment où la notification a été créée par le backend. Il pourrait permettre au frontend d'indiquer une fourchette horaire lors du `GET /v2/nofications?from=00000000&to=000000000`
+
+
+
+#### Exemples de notifications :
+
+##### notification de réception de email(s) :
+
+```yaml  
+notifications:  
+  - from: email-facility  
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: event  
+    timestamp: 1518691674517  
+    body:  
+      emailReceived: xxxxxx-xxxxx-xxxxx // uuid of new email  
+  - from: email-facility  
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: event  
+    timestamp: 1518691674517  
+    body:  
+      emailReceived: xxxxxx-xxxxx-xxxxx // uuid of new email  
+  - from: email-facility  
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: event  
+    timestamp: 1518691674517  
+    body:  
+      emailReceived: xxxxxx-xxxxx-xxxxx // uuid of new email  
+```
+
+##### notification de fin d'import de vCards :
+
+le frontend a initié un import avec un POST  sur /v1/imports
+
+lors de son prochain call sur GET /v2/notifications, il pourrait recevoir la notification suivante :
+
+```yaml  
+notifications:  
+  - from: contact-facility  
+    to: browser-notification  
+    id: xxxxxx-xxxxx-xxxxx  
+    type: feedback  
+    reference: xxxxxxxxx     // could be a hash of the initial call to the API (POST /v1/imports + timestamp (+ headers ?))  
+    timestamp: 1518691674517  
+    body: success  
+```
+
+##### notification d'échec d'envoi de email :
+
+l'utilisateur a envoyé un email (`POST /messages/{message_id}/actions`)
+
+lors de son prochain call sur GET /v2/notifications, le frontend pourrait recevoir la notification suivante :
+
+```yaml  
+notifications:  
+  - from: email-facility  
+    to: message-component  
+    id: xxxxx-xxxxx-xxxxx  
+    type: warning  
+    reference: xxxxxx-xxxxx-xxxx // uuid of email sent  
+    timestamp: 1518691674517  
+    body:  
+      warning: sending email failed  
+      code: 23  
+      message: MTA timeout  
+```
+
+##### notification que l'index est à jour (suite à create/update de document)
+
+l'utilisateur a modifié un contact (`PATCH /contacts/{contact_id}`)
+
+lors de son prochain call sur GET /v2/notifications, le frontend pourrait recevoir la notification suivante :
```

Comment:
-> utile pour invalider les données cachées dans le store redux 
    je verrais plutôt `to : entity-store, type: event`

    "event-manager" c'est trop abstrait